src/back/interfaces/handlers.py

```diff
@@ -0,0 +1,14 @@
+"""Module containing handlers' interfaces."""
+from typing import Generic, Self, TypeVar
+
+from src.back.interfaces.io import WriteStreamInterface
+
+T = TypeVar("T")
+
+
+class MessageHandlerInterface(Generic[T]):
```

Comment:
2) Here, I am making an interface "generic" meaning that it will take a type of a message to work with in its subclasses

Reply:
Yes. You also need to make it a `Protocol`